src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -9396,9 +9396,6 @@ void Unit::SetMaxHealth(uint32 val)
 
 void Unit::SetPower(Powers power, uint32 val, bool withPowerUpdate /*= true*/)
 {
-    if (GetPower(power) == val)
```

Comment:
this is a hackfix. This is forcing update fields to be updated despite already being up to date.

Reply:
> this is a hackfix. This is forcing update fields to be updated despite already being up to date.

At a low server update time diff, it often occurs that between two server ticks, the player's energy has only managed to regenerate by a fraction of a point (for example, 0.4 point). This change is stored in a special buffer (see `Player::Regenerate`), but the integer value of energy remains unchanged. Because of this, `Unit::SetPower` blocks  sending a packet with a power update to the client, and it becomes necessary to wait another 2 seconds, and the situation could repeat again. As a result, the client does not receive a power update for an extended period, even though theoretically it should receive one every 2 seconds.

Reply:
What about introducing another parameter for this function that allows this behavior only if called from Player::Regenerate?

Reply:
> What about introducing another parameter for this function that allows this behavior only if called from Player::Regenerate?

Done